src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs

```diff
@@ -693,4 +695,21 @@ void IPooledStream.DisposeCore()
     }
 
     long IPooledStream.PoolExpirationTicks => DrainExpirationTicks;
+
+    internal void ValidateHeaderValue(ReadOnlySpan<byte> name, ReadOnlySpan<byte> value)
+    {
+        // Validate the value doesn't contain newline characters.
+        // Doing this once when a value is added to the dynamic table avoids the need to repeat it every request.
+        if (ReferenceEquals(KestrelServerOptions.DefaultHeaderEncodingSelector, HttpRequestHeaders.EncodingSelector))
+        {
+            if (value.IndexOfAny((byte)'\r', (byte)'\n') >= 0)
+            {
+                throw new InvalidOperationException("Newline characters (CR/LF) are not allowed in request headers.");
+            }
```

Comment:
return?

Reply:
Opps! Yes.

(I haven't properly unit tested this code. I wanted to get an overall reaction before polishing)